### ISSUE TITLE
Kaniko pull flag equivalent

### ISSUE THREAD

**Issue by bensolom** [GoogleContainerTools/kaniko#2341]
Hello. 
Im using Dockerfile contains the first line with the convention
"FROM base_image_path:**latest**"
The base image is often updating its **latest** tag.
I want to ensure that my build is completely rebuild, including the **base image tag latest is not cached**.

In Docker, --no-cache flag was not sufficient as it forces rebuilding layers already available, **not** including the "FROM" line.
Therefore, in Docker I used the --pull flag that triggers a pull of the base image referenced using FROM and insuring Im getting the latest version.
Is there any Kaniko equivalent to the --pull flag or any other way to insure base image is not cached?

**Actual behavior**
Tried using --cache=false flag and added debug prints.

/kaniko/executor '--skip-tls-verify=true' '--cache=false' '--dockerfile=Dockerfile' --cache-copy-layers --push-retry 6 --use-new-run --cleanup --verbosity debug

Im not sure how to determine if the image with the tag "latest" was cached or not.
Im getting those logs messages:

Retrieving image manifest base_image_path:latest 
Retrieving image base_image_path:latest from registry harbor_path
Built cross stage deps: map[] 
Retrieving image manifest base_image_path:latest 
Returning cached image manifest   

**Expected behavior**
Not expecting to get the print "Returning cached image manifest"
According to [kaniko cache doc](https://cloud.google.com/build/docs/optimize-builds/kaniko-cache) it seems that the FROM
line is cached, thats why I tried using the --cache=false flag.
But those prints are confusing and im not sure what is actually happens

**To Reproduce**
1. Use the FROM line as the above
2. Use Kaniko executor as the above

 
 | **Description** | **Yes/No** |
 |----------------|---------------|
 | Please check if this a new feature you are proposing        | <ul><li>- [ No] </li></ul>|
 | Please check if the build works in docker but not in kaniko | <ul><li>- [ Yes] </li></ul>| 
 | Please check if this error is seen when you use `--cache` flag | <ul><li>- [ Yes] </li></ul>|
 | Please check if your dockerfile is a multistage dockerfile | <ul><li>- [ No] </li></ul>|

**Comment by bensolom** [comment]
UP

**Comment by bensolom** [comment]
UP

**Comment by GeorchW** [comment]
I had the same question. Looking [at the code](https://github.com/GoogleContainerTools/kaniko/blob/971b9f48a0cd9886f048601275aad261b4d22063/pkg/image/remote/remote.go#L34), it seems like the line 

```
Retrieving image base_image_path:latest from registry harbor_path
```

means that the image is actually being retrieved, and a later line

```
Returning cached image manifest
```

just means that this function is called twice for some reason, with the second call returning the then-cached version, if I understand it correctly.

---

However, I noticed that unpacking the rootfs happens within under 2s after retrieving the image:

```
INFO[0001] Unpacking rootfs as cmd [...] requires it. 
```

The image is way too big for that, which led me to the assumption that the layers are downloaded on-demand when being unpacked, and the steps before are just retrieving the layers.

Again, the code seems to support this assumption; I don't know it well though, so take my thoughts with a grain of salt. The images retrieved are `remoteImage`s, and I believe the layers are thus [`remoteLayer`](https://github.com/google/go-containerregistry/blob/03b86570e60f261bf4a79b903d73e03cb862910b/pkg/v1/remote/layer.go)s. These seem to only download the layer content when requested.

---

**TLDR:** The stuff you see in the logs is spurious, the image manifest is not cached between runs. The layers themselves are downloaded on-demand when unpacking the file system.
